podpac/core/algorithm/generic.py

```diff
@@ -33,6 +33,9 @@ def _inputs(self):
 
     def _first_init(self, **kwargs):
         trait_names = self.trait_names()
+        for key in kwargs:
+            if key in trait_names and isinstance(kwargs[key], Node):
+                raise RuntimeError("Trait '%s' is reserved and cannot be used as an Generic Algorithm input" % key)
```

Comment:
Oops, this particular check is supposed to be in develop already. If we decide we don't like this idea I'll make sure to commit it and its to develop anyways.